src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory_Methods.vb

```diff
@@ -516,6 +518,66 @@ Namespace Microsoft.CodeAnalysis.Operations
             End If
         End Function
 
+        Private Enum NestedType
+            None
+            DelegateCreation
+            ExpressionTree
+        End Enum
+
+        Private Shared Function GetNestedInfo(operand As BoundNode, conversionKind As ConversionKind, conversionSyntax As SyntaxNode, targetType As TypeSymbol
+                                                                         ) As (NestedType As NestedType, NestedConversionKind As ConversionKind)
+            If conversionKind = ConversionKind.Identity AndAlso operand.Kind = BoundKind.Parenthesized Then
+                Dim nestedOperand = operand
+                Do While nestedOperand.Kind = BoundKind.Parenthesized
+                    nestedOperand = DirectCast(nestedOperand, BoundParenthesized).Expression
+                Loop
+
+                Select Case nestedOperand.Kind
+                    Case BoundKind.Conversion, BoundKind.DirectCast, BoundKind.TryCast
+                        Dim nestedConversion = DirectCast(nestedOperand, BoundConversionOrCast)
+                        If nestedConversion.Type <> targetType Then
+                            Return (NestedType.None, NestedConversionKind:=Nothing)
+                        End If
+
+                        If IsDelegateCreation(nestedConversion.ConversionKind, conversionSyntax, nestedConversion.Operand, targetType) Then
+                            Return (NestedType.DelegateCreation, nestedConversion.ConversionKind)
+                        ElseIf (nestedConversion.ConversionKind And ConversionKind.ConvertedToExpressionTree) = ConversionKind.ConvertedToExpressionTree Then
+                            Return (NestedType.ExpressionTree, nestedConversion.ConversionKind)
+                        End If
+                    Case BoundKind.DelegateCreationExpression
+                        Dim nestedDelegateCreation = DirectCast(nestedOperand, BoundDelegateCreationExpression)
+                        If nestedDelegateCreation.Type = targetType Then
+                            Return (NestedType.DelegateCreation, NestedConversionKind:=conversionKind)
+                        End If
+                End Select
+            End If
+            Return (NestedType.None, NestedConversionKind:=Nothing)
+        End Function
+
+        Private Function CreateNestedDelegateCreationOrExpressionTreeOperand(operand As BoundNode, conversionSyntax As SyntaxNode) As Lazy(Of IOperation)
+            Select Case operand.Kind
+                Case BoundKind.Parenthesized
+                    Dim boundParenthesized = DirectCast(operand, BoundParenthesized)
+                    ' Because we're in a delegate creation scenario, the underlying expression actually doesn't have a type.
+                    Dim type As ITypeSymbol = Nothing
```

Comment:
>Dim type As ITypeSymbol = Nothing [](start = 20, length = 33)

I think it would be more robust to simply inherit the type from the operand, i.e. result of the nested call to CreateNestedDelegateCreationOrExpressionTreeOperand.

Reply:
I guess that would be complicated because the returned value is Lasy.

---
In reply to: [153646656](https://github.com/dotnet/roslyn/pull/23315#discussion_r153646656) [](ancestors = 153646656)

Reply:
I think I am going to have to do this, as my new tests with nested conversions under nested conversions shouldn't have `null` as the type of the parenthesized operation.

---
In reply to: [153654654](https://github.com/dotnet/roslyn/pull/23315#discussion_r153654654) [](ancestors = 153654654,153646656)